platform/linux-generic/odp_packet.c

```diff
@@ -816,10 +816,7 @@ odp_packet_t odp_packet_alloc(odp_pool_t pool_hdl, uint32_t len)
 	odp_packet_t pkt;
 	int num, num_seg;
 
-	if (odp_unlikely(pool->params.type != ODP_POOL_PACKET)) {
-		__odp_errno = EINVAL;
-		return ODP_PACKET_INVALID;
-	}
+	ODP_ASSERT(pool->params.type == ODP_POOL_PACKET);
```

Comment:
I think this is not correct. Before if we detected an error we interrupted the function returning an error code, now we will only interrupt this function (by calling abort_fn() global function) if and only if ODP was compiled with ODP_DEBUG enabled.